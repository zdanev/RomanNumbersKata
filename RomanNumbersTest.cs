using System.Linq;

using Microsoft.VisualStudio.TestTools.UnitTesting;

using ApprovalTests;
using ApprovalTests.Reporters;

namespace RomanNumbers
{
    [TestClass]
    [UseReporter(typeof(DiffReporter))]
    public class RomanNumbersTest
    {
        [TestMethod]
        public void Test_I_1()
        {
            Assert.AreEqual(1, Roman.ToDecimal("I"));
        }

        [TestMethod]
        public void TestAll()
        {
            var romans = new[] {"I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX", "X"};
            var results = romans.Select(r => new {roman = r, number = Roman.ToDecimal(r)});
            Approvals.VerifyAll(results, "");
        }
    }
}