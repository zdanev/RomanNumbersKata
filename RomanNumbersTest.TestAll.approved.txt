[0] = { roman = I, number = 1 }
[1] = { roman = II, number = 2 }
[2] = { roman = III, number = 3 }
[3] = { roman = IV, number = 4 }
[4] = { roman = V, number = 5 }
[5] = { roman = VI, number = 6 }
[6] = { roman = VII, number = 7 }
[7] = { roman = VIII, number = 8 }
[8] = { roman = IX, number = 9 }
[9] = { roman = X, number = 10 }
